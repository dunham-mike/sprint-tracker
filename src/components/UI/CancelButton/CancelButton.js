import React from 'react';
import { withStyles } from '@material-ui/core/styles';

import red from '@material-ui/core/colors/red';
import CancelOutlinedIcon from '@material-ui/icons/CancelOutlined';

const styles = theme => ({
    cancelButtonContainer: {
        cursor: 'pointer',
        color: red[100],
        margin: theme.spacing(1),
        '&:hover': {
            color: red[200],
        },
    },
});

const cancelButton = (props) => {
    const { classes } = props;

    return (
        <div className={classes.cancelButtonContainer} onClick={props.clicked}>
            <CancelOutlinedIcon fontSize="large"/>
        </div>
    );
};

export default withStyles(styles)(cancelButton);