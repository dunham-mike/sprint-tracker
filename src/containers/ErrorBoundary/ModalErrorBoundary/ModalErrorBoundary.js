import React, { Component } from 'react';
import { withStyles } from '@material-ui/core/styles';

import Modal from '@material-ui/core/Modal';
import Paper from '@material-ui/core/Paper';
import Typography from '@material-ui/core/Typography';

import CancelButton from '../../../components/UI/CancelButton/CancelButton';

const styles = theme => ({
    modal: {
        display: 'flex',
        alignItems: 'center',
        justifyContent: 'center',
    },
    paper: {
        backgroundColor: theme.palette.background.paper,
        boxShadow: theme.shadows[5],
        padding: theme.spacing(5, 4, 5),
        outline: '0',
        width: theme.spacing(60),
        position: 'relative',
        zIndex: '1120',
        textAlign: 'center',
    },
    cancelButtonContainer: {
        position: 'absolute',
        top: '0',
        right: '0',
        // top: '7%',
        // right: '7%',
    },
});


// Modeled on: https://reactjs.org/docs/error-boundaries.html
class errorBoundary extends Component {
    constructor(props) {
      super(props);
      this.state = { hasError: false };
    }
  
    static getDerivedStateFromError(error) {
      // Update state so the next render will show the fallback UI.
      return { hasError: true };
    }
  
    // componentDidCatch(error, errorInfo) {
    //   // You can also log the error to an error reporting service
    //   logErrorToMyService(error, errorInfo);
    // }
  
    render() {
        const { classes } = this.props;

        if (this.state.hasError) {
            // Rendering custom fallback UI
            return (
                <Modal
                    aria-labelledby="simple-modal-title"
                    aria-describedby="simple-modal-description"
                    open={true}
                    className={classes.modal}
                    onClose={this.props.onClose}
                >
                    <Paper className={classes.paper}>
                        <div className={classes.cancelButtonContainer} >
                            <CancelButton clicked={this.props.onClose} />
                        </div>
                        <Typography variant="h5">
                            Something went wrong! Please try something else or refresh the page.
                        </Typography>
                    </Paper>
                </Modal>
            );
        }
    
        return this.props.children; 
    }
  }

  export default withStyles(styles)(errorBoundary);