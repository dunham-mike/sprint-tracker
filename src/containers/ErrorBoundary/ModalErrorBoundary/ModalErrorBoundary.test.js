import React from 'react';
import Enzyme, { mount } from 'enzyme'
import Adapter from 'enzyme-adapter-react-16'
import toJson from 'enzyme-to-json';

import ModalErrorBoundary from './ModalErrorBoundary';
import CancelButton from '../../../components/UI/CancelButton/CancelButton';

Enzyme.configure({ adapter: new Adapter() });

const mountSetup = () => {

    const enzymeMountWrapper = mount(
        <ModalErrorBoundary>
            <CancelButton />
        </ModalErrorBoundary>
    );

    return {
        enzymeMountWrapper
    }
}

describe('ModalErrorBoundary Component', () => {

    it('should match the snapshot', () => {
        const { enzymeMountWrapper } = mountSetup();
        expect(toJson(enzymeMountWrapper)).toMatchSnapshot();
    });

    it('should display an ErrorMessage if wrapped component throws', () => {  
        const { enzymeMountWrapper } = mountSetup();

        const error = new Error('test_error');
    
        enzymeMountWrapper.find(CancelButton).simulateError(error);
    
        expect(enzymeMountWrapper.find('.MuiTypography-root.MuiTypography-h5').prop('children'))
            .toEqual('Something went wrong! Please try something else or refresh the page.');
    });

});