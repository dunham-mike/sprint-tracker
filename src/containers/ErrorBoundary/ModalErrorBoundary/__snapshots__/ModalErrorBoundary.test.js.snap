// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ModalErrorBoundary Component should match the snapshot 1`] = `
<WithStyles(errorBoundary)>
  <errorBoundary
    classes={
      Object {
        "cancelButtonContainer": "errorBoundary-cancelButtonContainer-3",
        "modal": "errorBoundary-modal-1",
        "paper": "errorBoundary-paper-2",
      }
    }
  >
    <WithStyles(cancelButton)>
      <cancelButton
        classes={
          Object {
            "cancelButtonContainer": "cancelButton-cancelButtonContainer-4",
          }
        }
      >
        <div
          className="cancelButton-cancelButtonContainer-4"
        >
          <ForwardRef
            fontSize="large"
          >
            <WithStyles(ForwardRef(SvgIcon))
              fontSize="large"
            >
              <ForwardRef(SvgIcon)
                classes={
                  Object {
                    "colorAction": "MuiSvgIcon-colorAction",
                    "colorDisabled": "MuiSvgIcon-colorDisabled",
                    "colorError": "MuiSvgIcon-colorError",
                    "colorPrimary": "MuiSvgIcon-colorPrimary",
                    "colorSecondary": "MuiSvgIcon-colorSecondary",
                    "fontSizeInherit": "MuiSvgIcon-fontSizeInherit",
                    "fontSizeLarge": "MuiSvgIcon-fontSizeLarge",
                    "fontSizeSmall": "MuiSvgIcon-fontSizeSmall",
                    "root": "MuiSvgIcon-root",
                  }
                }
                fontSize="large"
              >
                <svg
                  aria-hidden="true"
                  className="MuiSvgIcon-root MuiSvgIcon-fontSizeLarge"
                  focusable="false"
                  role="presentation"
                  viewBox="0 0 24 24"
                >
                  <path
                    d="M12 2C6.47 2 2 6.47 2 12s4.47 10 10 10 10-4.47 10-10S17.53 2 12 2zm0 18c-4.41 0-8-3.59-8-8s3.59-8 8-8 8 3.59 8 8-3.59 8-8 8zm3.59-13L12 10.59 8.41 7 7 8.41 10.59 12 7 15.59 8.41 17 12 13.41 15.59 17 17 15.59 13.41 12 17 8.41z"
                  />
                </svg>
              </ForwardRef(SvgIcon)>
            </WithStyles(ForwardRef(SvgIcon))>
          </ForwardRef>
        </div>
      </cancelButton>
    </WithStyles(cancelButton)>
  </errorBoundary>
</WithStyles(errorBoundary)>
`;
